src/main/java/frc/robot/commands/ReefLateralAlignment.java

```diff
@@ -0,0 +1,137 @@
+ */
+ 
+package frc.robot.commands;
+
+import com.nrg948.preferences.RobotPreferences.DoubleValue;
+import com.nrg948.preferences.RobotPreferencesValue;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants.VisionConstants;
+import frc.robot.commands.AlignToReef.ReefPosition;
+import frc.robot.subsystems.AprilTag;
+import frc.robot.subsystems.Subsystems;
+import frc.robot.subsystems.Swerve;
+import frc.robot.util.FieldUtils;
+import java.util.Optional;
+import org.photonvision.targeting.PhotonTrackedTarget;
+
+/* You should consider using the more terse Command factories API instead https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html#defining-commands */
+public class ReefLateralAlignment extends Command {
+  private static final double SCALE_FACTOR = 0.3;
+
+  @RobotPreferencesValue
+  public static DoubleValue Pr = new DoubleValue("AlignToPose", "Lateral R KP", 5.0);
+
+  @RobotPreferencesValue
+  public static DoubleValue Py = new DoubleValue("AlignToPose", "Lateral Y KP", 3.0);
+
+  private Swerve drivetrain;
+  private AprilTag vision;
+  private int targetID;
+  private ProfiledPIDController yController;
+  private ProfiledPIDController rController;
+  private ReefPosition reefPosition;
+  private boolean isAligningY = false;
+  private ChassisSpeeds chassisSpeeds = new ChassisSpeeds();
+
+  /** Creates a new TagLateralAlignment. */
+  public ReefLateralAlignment(Subsystems subsystems, ReefPosition reefPosition) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    drivetrain = subsystems.drivetrain;
+    vision = subsystems.frontCamera.get();
+    this.reefPosition = reefPosition;
+
+    yController =
+        new ProfiledPIDController(
+            Py.getValue(),
+            0,
+            0,
+            new TrapezoidProfile.Constraints(
+                Swerve.getMaxSpeed() * SCALE_FACTOR, Swerve.getMaxAcceleration() * SCALE_FACTOR));
+    rController =
+        new ProfiledPIDController(
+            Math.toRadians(Pr.getValue()), 0, 0, Swerve.getRotationalConstraints());
+    addRequirements(drivetrain);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    targetID = vision.getBestTarget().getFiducialId(); // bad, but good enough for now?
+    double targetY;
+    switch (reefPosition) {
+      case LEFT_BRANCH:
+        targetY = -VisionConstants.BRANCH_TO_REEF_APRILTAG;
+        break;
+
+      case CENTER_REEF:
+        targetY = 0;
+        break;
+
+      case RIGHT_BRANCH:
+        targetY = VisionConstants.BRANCH_TO_REEF_APRILTAG;
+        break;
+
+      default:
+        targetY = 0;
+        break;
+    }
```

Comment:
This suggestion assumes the `ReefPosition` enum is refactored as suggested in a different PR.
```suggestion
    double targetY = reefPosition.yOffset();
```

Reply:
See https://github.com/NRG948/NRGRobot2025/pull/176